Header header
QMatrixRow[] q_matrix